rewrite-gradle/src/main/java/org/openrewrite/gradle/EnableDevelocityBuildCache.java

```diff
@@ -0,0 +1,130 @@
+            description = "Value for `//develocity/buildCache/remote/storeEnabled`.",
+            example = "true",
+            required = false)
+    @Nullable
+    String remotePushEnabled;
+
+    @Override
+    public Validated<Object> validate(ExecutionContext ctx) {
+        return super.validate(ctx)
+                .or(Validated.notBlank("remoteEnabled", remoteEnabled)
+                        .or(Validated.notBlank("remotePushEnabled", remotePushEnabled)));
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(new IsSettingsGradle<>(),
+                new GroovyIsoVisitor<ExecutionContext>() {
+                    @Override
+                    public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+                        J.MethodInvocation m = super.visitMethodInvocation(method, ctx);
+
+                        if ("buildCache".equals(method.getSimpleName())) {
+                            try {
+                                Cursor parent = getCursor().dropParentUntil(v -> v instanceof J.MethodInvocation && "develocity".equals(((J.MethodInvocation) v).getSimpleName()));
+                                parent.putMessage("hasBuildCacheConfig", true);
+                            } catch (IllegalStateException e) {
+                                // ignore, this means we're not in a develocity block
+                            }
+                            return m;
+                        }
+
+                        if ("develocity".equals(method.getSimpleName()) && getCursor().pollMessage("hasBuildCacheConfig") != null) {
+                            J.MethodInvocation buildCache = develocityBuildCacheTemplate("    ", ctx);
+
+                            return maybeAutoFormat(m, m.withArguments(ListUtils.mapFirst(m.getArguments(), arg -> {
+                                if (arg instanceof J.Lambda) {
+                                    J.Lambda lambda = (J.Lambda) arg;
+                                    J.Block block = (J.Block) lambda.getBody();
+                                    return lambda.withBody(block.withStatements(ListUtils.concat(block.getStatements(), buildCache)));
+                                }
+                                return arg;
+                            })), ctx);
+                        }
+                        return m;
+                    }
+                }
+        );
+    }
+
+    private J.@Nullable MethodInvocation develocityBuildCacheTemplate(String indent, ExecutionContext ctx) {
+        StringBuilder ge = new StringBuilder("\ndevelocity {\n");
+        ge.append(indent).append("buildCache {\n");
+
+        ge.append(indent).append(indent).append("remote(develocity.buildCache) {\n");
+        if (!StringUtils.isBlank(remoteEnabled)) {
+            ge.append(indent).append(indent).append(indent).append("enabled = ").append(remoteEnabled).append("\n");
+        }
```

Comment:
```suggestion
        String ge =
            "\ndevelocity {\n" +
            "    buildCache {\n " +
            "        develocity.buildCache) {\n";
         if (!StringUtils.isBlank(remoteEnabled)) {
             ge += "            enabled = " + remoteEnabled + "\n";
         }    
```
etc...

is way easier to read, as you can see the template 'nearly' as code.